cmake/Modules/FindAubio.cmake

```diff
@@ -1,21 +1,25 @@
-if (NOT USE_SELF_BUILT_AUBIO)
-    find_package(PkgConfig REQUIRED)
-    if(PKG_CONFIG_FOUND)
-        # The version we look for is legacy, may need some fine tunning
-        pkg_check_modules(AUBIO IMPORTED_TARGET GLOBAL aubio>=0.4.9)
-            if(AUBIO_FOUND)
-                add_library(aubio ALIAS PkgConfig::AUBIO)
-            endif()
-    endif()
-endif()
+include(LibFindMacros)
+
+set(Aubio_FIND_REQUIRED FALSE)
+set(Aubio_FIND_QUIETLY TRUE)
+libfind_pkg_detect(Aubio aubio FIND_PATH aubio.h PATH_SUFFIXES aubio FIND_LIBRARY aubio)
+libfind_process(Aubio)
 
-if (NOT AUBIO_FOUND)
-    message("Building aubio from sources...")
+if (Aubio_FOUND)
+	set(Aubio_VERSION ${Aubio_PKGCONF_VERSION})
+else ()
+    message("-- Fetching aubio from github...")
     include(FetchContent)
-    FetchContent_Declare(aubio-sources
+    set(Aubio_VERSION "0.4.9-performous")
+    FetchContent_Declare(aubio_from_source
       GIT_REPOSITORY https://github.com/performous/aubio.git
+      GIT_SHALLOW    TRUE
       GIT_TAG        14fec3da6749fbcc47b56648d7a38296eccd9499
       SOURCE_DIR aubio-src
     )
-    FetchContent_MakeAvailable(aubio-sources)
+    FetchContent_MakeAvailable(aubio_from_source)
+    set(Aubio_INCLUDE_DIRS ${aubio_from_source_SOURCE_DIR}/include/)
+    set(Aubio_LIBRARIES "-laubio")
```

Comment:
you should  not use variables for targets, this is really discouraged by cmake

Reply:
I revert back to former code style to detect Aubio system installation